dispatcher/backend/src/common/schemas/offliners/devdocs.py

```diff
@@ -0,0 +1,162 @@
+from marshmallow import fields
+
+from common.schemas import SerializableSchema, String
+from common.schemas.fields import (
+    validate_output,
+    validate_zim_description,
+    validate_zim_longdescription,
+)
+
+
+class DevDocsFlagsSchema(SerializableSchema):
+    class Meta:
+        ordered = True
+
+    all_flag = fields.Boolean(
+        truthy=[True],
+        falsy=[False],
+        metadata={
+            "label": "All",
+            "description": "Fetch all Devdocs resources, and produce one ZIM "
+            "per resource.",
+        },
+        data_key="all",
+    )
+
+    slug = String(  # should be ListOfString but not yet supported by Zimfarm
+        metadata={
+            "label": "Slug",
+            "description": "Fetch the provided Devdocs resource. "
+            "Slugs are the first path entry in the Devdocs URL. "
+            "For example, the slug for: `https://devdocs.io/gcc~12/` is `gcc~12`.",
+        },
+    )
+
+    first = fields.Integer(
+        metadata={
+            "label": "Number of first items",
+            "description": "Fetch only the first N items per slug as shown "
+            "in the DevDocs UI. Do not set to fetch all items.",
+        },
+    )
+
+    skip_slug_regex = String(
+        metadata={
+            "label": "Skip slugs regex",
+            "description": "Skips slugs matching the given regular expression."
+            "Do not set to fetch all slugs",
+        },
+        data_key="skip-slug-regex",
+    )
+
+    file_name_format = String(
+        metadata={
+            "label": "ZIM filename",
+            "description": "ZIM filename. Do not input trailing `.zim`, it "
+            "will be automatically added. You can use placeholders, see "
+            "https://github.com/openzim/devdocs/blob/main/README.md. Defaults "
+            "to devdocs.io_en_{clean_slug}_{period}",
+        },
+        data_key="file-name-format",
+    )
+
+    name_format = String(
```

Comment:
I thought we wanted to get away from those auto-generated metadata?

Reply:
I think we want to avoid auto-generated metadata which might contain only garbage, be too long, ...

Here we know (and have tested) that these metadata are compliant with all existing slugs on devdocs.io, and it allows to ensure some homogeneity (avoiding typos typically) so I think they make much more sense than description automatically grabbed from Youtube channel typically.

That being said, I can hide (not document) them on the Zimfarm if it seems better.